src/SFML/Window/Win32/WindowImplWin32.cpp

```diff
@@ -1003,14 +1021,78 @@ void WindowImplWin32::processEvent(UINT message, WPARAM wParam, LPARAM lParam)
             if ((wParam == DBT_DEVICEARRIVAL) || (wParam == DBT_DEVICEREMOVECOMPLETE))
             {
                 // Some sort of device change has happened, update joystick connections if it is a device interface
-                auto* deviceBroadcastHeader = reinterpret_cast<DEV_BROADCAST_HDR*>(lParam);
+                const auto* deviceBroadcastHeader = reinterpret_cast<DEV_BROADCAST_HDR*>(lParam);
 
                 if (deviceBroadcastHeader && (deviceBroadcastHeader->dbch_devicetype == DBT_DEVTYP_DEVICEINTERFACE))
                     JoystickImpl::updateConnections();
             }
 
             break;
         }
+        case WM_TOUCH:
+        {
+            // Get the number of events
+            const unsigned int count = LOWORD(wParam);
+
+            // Reserve memory
+            std::vector<TOUCHINPUT> events(count);
+
+            if (GetTouchInputInfo(reinterpret_cast<HTOUCHINPUT>(lParam), count, events.data(), sizeof(TOUCHINPUT)))
+            {
+                POINT point;
+                for (unsigned int i = 0; i < count; ++i)
+                {
+                    Event event;
+                    int   index = getTouchID(events[i].dwID);
+
+                    // Out of IDs? Should never happen
+                    if (index == -1)
+                        continue;
+
+                    event.touch.finger = static_cast<unsigned int>(index);
```

Comment:
```suggestion
                    const int index = getTouchID(events[i].dwID);

                    // Out of IDs? Should never happen
                    if (index == -1)
                        continue;

                    Event event;
                    event.touch.finger = static_cast<unsigned int>(index);
```
I'd prefer if we not declare a variable until right before it's needed. It helps local reasoning.